### PR TITLE
Add a combined non-linear Gaussian measurement model 

### PR DESCRIPTION
**Pull request by sdhiscocks**
This allows multiple measurement models to be combined together, in particular for combining a non-linear model with one or more, linear or non-linear models.

### CODE REVIEW

File: stonesoup/models/measurement/nonlinear.py [hunk 1]
```diff
@@ -1,19 +1,97 @@
...
+
+        return state_vector
+
+    def covar(self, **kwargs):
+        return block_diag(
+            *(model.covar(**kwargs) for model in self.model_list)
+            ).view(CovarianceMatrix)
+
+    def rvs(self, num_samples=1, **kwargs):
+        rvs_vectors = sp.vstack([model.rvs(num_samples, **kwargs)
```
Comment (etfrogers-dstl) [inline stonesoup/models/measurement/nonlinear.py]: np.vstack?

File: stonesoup/models/measurement/nonlinear.py [hunk 2]
```diff
@@ -168,9 +246,7 @@ def function(self, state_vector, noise=None, **kwargs):
         # Convert to Spherical
         rho, phi, theta = cart2sphere(*xyz_rot[:, 0])
 
-        return sp.array([[Elevation(theta)],
```
Comment (etfrogers-dstl) [inline stonesoup/models/measurement/nonlinear.py]: np.array

File: stonesoup/models/measurement/nonlinear.py [hunk 3]
```diff
@@ -183,7 +259,7 @@ def inverse_function(self, state_vector, **kwargs):
         xyz = [xyz_rot[0][0], xyz_rot[1][0], xyz_rot[2][0]]
         x, y, z = xyz + self.translation_offset[:, 0]
 
-        res = sp.zeros((self.ndim_state, 1))
+        res = sp.zeros((self.ndim_state, 1)).view(StateVector)
```
Comment (etfrogers-dstl) [inline stonesoup/models/measurement/nonlinear.py]: np.array

File: stonesoup/models/measurement/nonlinear.py [hunk 4]
```diff
@@ -283,7 +359,7 @@ def inverse_function(self, state_vector, **kwargs):
         xy = [xyz_rot[0][0], xyz_rot[1][0]]
         x, y = xy + self.translation_offset[:, 0]
 
-        res = sp.zeros((self.ndim_state, 1))
```
Comment (etfrogers-dstl) [inline stonesoup/models/measurement/nonlinear.py]: np.zeros

**Comment by sdhiscocks** [comment]
I've rebased this onto _master_ to account for changes with `np`/`sp` usage.

**Pull request merged**